controllers/memberstatus/memberstatus_controller_test.go

```diff
@@ -668,6 +674,122 @@ func TestOverallStatusCondition(t *testing.T) {
+		requestName := defaultMemberStatusName
+		memberOperatorDeployment := newMemberDeploymentWithConditions(status.DeploymentAvailableCondition(), status.DeploymentProgressingCondition())
+		memberStatus := newMemberStatus()
+		getHostClusterFunc := newGetHostClusterReady
+		// we have a secret that contains the access token for GitHub authenticated APIs
+		githubSecret := test.CreateSecret("github", test.MemberOperatorNs, map[string][]byte{
+			"accessToken": []byte("abcd1234"),
+		})
+		commitTimeStamp := time.Now().Add(-time.Hour * 1)
+		latestCommitSHA := "xxxxaaaaa"  // we set the latest commit to something that differs from the `buildCommitSHA` constant
+		version.Commit = buildCommitSHA // let's set the build version to a constant value
+		reconciler, req, fakeClient := prepareReconcile(t, requestName, getHostClusterFunc, allNamespacesCl, mockLastGitHubAPICall, test.MockGitHubClientForRepositoryCommits(latestCommitSHA, commitTimeStamp), append(nodeAndMetrics, memberOperatorDeployment, memberStatus, config, githubSecret)...)
+
+		// when
+		res, err := reconciler.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, requeueResult, res)
+		AssertThatMemberStatus(t, req.Namespace, requestName, fakeClient).
+			HasCondition(ComponentsNotReady("memberOperator")).
+			HasMemberOperatorConditions(ConditionReady(toolchainv1alpha1.ToolchainStatusDeploymentReadyReason)).
+			HasMemberOperatorRevisionCheckConditions(ConditionNotReady(toolchainv1alpha1.ToolchainStatusDeploymentNotUpToDateReason, "deployment version is not up to date with latest github commit SHA. deployed commit SHA "+version.Commit+" ,github latest SHA "+latestCommitSHA+", expected deployment timestamp: "+commitTimeStamp.Add(status.DeploymentThreshold).Format(time.RFC3339))).
+			HasMemoryUsage(OfNodeRole("master", 33), OfNodeRole("worker", 25)).
+			HasRoutes("https://console.member-cluster/console/", "https://codeready-codeready-workspaces-operator.member-cluster/che/", routesAvailable())
+	})
+
+	t.Run("member operator deployment version is up to date", func(t *testing.T) {
+		// given
+		config := commonconfig.NewMemberOperatorConfigWithReset(t, testconfig.MemberEnvironment("prod"), testconfig.MemberStatus().GitHubSecretRef("github").GitHubSecretAccessTokenKey("accessToken"))
+		requestName := defaultMemberStatusName
+		memberOperatorDeployment := newMemberDeploymentWithConditions(status.DeploymentAvailableCondition(), status.DeploymentProgressingCondition())
+		memberStatus := newMemberStatus()
+		getHostClusterFunc := newGetHostClusterReady
+		// we have a secret that contains the access token for GitHub authenticated APIs
+		githubSecret := test.CreateSecret("github", test.MemberOperatorNs, map[string][]byte{
+			"accessToken": []byte("abcd1234"),
+		})
```

Comment:
the given parts in the test seems to be the same - could we move it to a parent test so it could be reused?

Reply:
I've created _"member operator deployment revision check"_ as a parent for all the revision check unit tests.
TBH I tried putting also the `devConfig` and `prodConfig` there but the tests were randomly failing , I wasn't yet able to understand why, the reconciler and fakeClient are scoped to each specific test, so not sure what's happening. I've tried also moving the `memberStatus` to each test but still not working. 

Long story short - I had to create this `devConfig` and `prodConfig` in each subtest.

Plz see [78a30e1](https://github.com/codeready-toolchain/member-operator/pull/446/commits/78a30e10315b864af392e7a11c435de06468920a) and [daee1ad](https://github.com/codeready-toolchain/member-operator/pull/446/commits/daee1ad8962b8530c1da5e096e0f3ac12a8b7a1d)